From 12b9d20a5fe7847b4f623ff1ccef3c588b776a77 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Thu, 22 Aug 2013 14:50:09 +0800
Subject: [PATCH 20/32] ENGR00275869 Audio output speed for FFx2 is same with
 that for FFx1.5

result speed of setPlaySpeed needs to return to caller, to judge
whether this call success or failed.

Change-Id: I2ecfc59a5f6fe9a9910bc7d43b4e4b692c2d603a
Signed-off-by: Xu Mao <B34688@freescale.com>
---
 api/current.txt                           | 2 +-
 media/java/android/media/MediaPlayer.java | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/api/current.txt b/api/current.txt
index f6ccc18..e0f5c69 100644
--- a/api/current.txt
+++ b/api/current.txt
@@ -12579,7 +12579,7 @@ package android.media {
     method public void setOnSeekCompleteListener(android.media.MediaPlayer.OnSeekCompleteListener);
     method public void setOnTimedTextListener(android.media.MediaPlayer.OnTimedTextListener);
     method public void setOnVideoSizeChangedListener(android.media.MediaPlayer.OnVideoSizeChangedListener);
-    method public void setPlaySpeed(int);
+    method public void setPlaySpeed(int[]);
     method public void setScreenOnWhilePlaying(boolean);
     method public void setSurface(android.view.Surface);
     method public void setVideoScalingMode(int);
diff --git a/media/java/android/media/MediaPlayer.java b/media/java/android/media/MediaPlayer.java
index e3f8690..817a214 100644
--- a/media/java/android/media/MediaPlayer.java
+++ b/media/java/android/media/MediaPlayer.java
@@ -1552,8 +1552,9 @@ public class MediaPlayer implements SubtitleController.Listener
      * When normalized speed is [0.1, 1.9] audio is outputted, otherwise audio
      * is not outputted.
      */
-    public void setPlaySpeed(int Speed) {
-        setParameter(KEY_PARAMETER_PLAYBACK_RATE_PERMILLE, Speed);
+    public void setPlaySpeed(int[] Speed) {
+        setParameter(KEY_PARAMETER_PLAYBACK_RATE_PERMILLE, Speed[0]);
+        Speed[0] = getIntParameter(KEY_PARAMETER_PLAYBACK_RATE_PERMILLE);
     }
 
     /*
-- 
1.8.0

